packages/augur-tools/src/flash/flash.ts

```diff
@@ -160,60 +160,76 @@ export class FlashSession {
-
-      // Create a ContractAPI for this user with this particular augur client. This provides
-      // a variety of nice wrapper functions which we should think about exporting
-      this.user = new ContractAPI(client, this.provider, client.dependencies, account);
-      this.user.augur.setGasPrice(new BigNumber(this.network.gasPrice.toString()));
-
-      // IF we want this flash client to use a safe associated with the past in
-      // account, configure it at this point.
-      if (config.gnosis.enabled) {
-        const safe = await this.user.getOrCreateSafe();
-        await this.user.faucetOnce(new BigNumber(1e21), safe);
-        const safeStatus = await this.user.getSafeStatus(safe);
-        console.log(`Safe ${safe}: ${safeStatus}`);
-        this.user.augur.setGasPrice(new BigNumber(90000));
-        this.user.setGnosisSafeAddress(safe);
-        this.user.setUseGnosisSafe(true);
-        this.user.setUseGnosisRelay(true);
-      } else if (approveCentralAuthority) {
-        await this.user.approveCentralAuthority();
+      try {
+        // Get an actual account for the provided public address. This also
+        // handles the case where none is passed in, in which case it will use
+        // the default account (0)
+        const account = this.getAccount(accountAddress);
+        console.log(`Account Address: ${account.publicKey}`);
+        // Within flash we want to use an account with a private key as the signer
+        // so we manually create our own signer here.
+        const signer = await makeSigner(account, this.provider);
+
+        // Run everything in one context, both syncing and this client code
+        const connector = new Connectors.SingleThreadConnector();
+        const client = await createClient(config, connector, account.publicKey, signer, this.provider);
+
+        // Create a ContractAPI for this user with this particular augur client. This provides
+        // a variety of nice wrapper functions which we should think about exporting
+        this.user = new ContractAPI(client, this.provider, client.dependencies, account);
+        this.user.augur.setGasPrice(new BigNumber(this.network.gasPrice.toString()));
+
+        // IF we want this flash client to use a safe associated with the past in
+        // account, configure it at this point.
+        if (config.gnosis.enabled) {
+          const safe = await this.user.getOrCreateSafe();
+          await this.user.faucetOnce(new BigNumber(1e21), safe);
+          const safeStatus = await this.user.getSafeStatus(safe);
+          console.log(`Safe ${safe}: ${safeStatus}`);
+          this.user.augur.setGasPrice(new BigNumber(90000));
+          this.user.setGnosisSafeAddress(safe);
+          this.user.setUseGnosisSafe(true);
+          this.user.setUseGnosisRelay(true);
+        } else if (approveCentralAuthority) {
+          await this.user.approveCentralAuthority();
+        }
+      } catch (e) {
+        throw e;
```

Comment:
is this necessary?

Reply:
Currently when you run scripts that call ensureUser it tosses random logs into the console and isn't really understandable.

Seems better to try to make this pretty while in here.